### PR TITLE
apptainer: new package

### PR DESCRIPTION
**Pull request by matz-e**
As Apptainer is just a renamed Singularity, use the structure of the
current recipes, replacing naming where appropriate.

### CODE REVIEW

File: var/spack/repos/builtin/packages/singularityce/package.py [hunk 1]
```diff
@@ -36,11 +44,11 @@ def gopath(self):
 
     @property
     def sylabs_gopath_dir(self):
-        return join_path(self.gopath, 'src/github.com/sylabs/')
+        return join_path(self.gopath, 'src', 'github.com', self.singularity_org)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/singularityce/package.py]: Maybe rename `sylabs_gopath_dir` since it's in the base package :grimacing: ?

**Review by alalazo — CHANGES_REQUESTED** [review]
One minor request

File: var/spack/repos/builtin/packages/singularityce/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/singularityce/package.py]: Also, should we do something about `singularity`?

Comment (matz-e) [reply var/spack/repos/builtin/packages/singularityce/package.py]: I was tempted to just remove that property and merge it with `singularity_gopath_dir`, as it's not used anywhere else. But I can also rename it, any suggestions? Or am I misunderstanding you?

W.r.t. `singularity`, it seems that the Github page states that it is EOL, in which case it should be deprecated. OTOH, there is a fairly recent release there, too.

Comment (alalazo) [reply var/spack/repos/builtin/packages/singularityce/package.py]: Removing the property is fine with me.

**Review by alalazo — APPROVED** [review]

**Pull request merged**